src/Components/Trainers/index.jsx

```diff
@@ -7,40 +10,45 @@ import SharedModal from '../Shared/Modal';
 
 const Trainers = () => {
   const history = useHistory();
-  const [trainers, setTrainers] = useState([]);
   const [showModal, setShowModal] = useState(false);
+  const [stateModal, setStateModal] = useState('success');
   const [modalMessage, setModalMessage] = useState('');
   const [selectedTrainerId, setSelectedTrainerId] = useState(null);
   const [isConfirmationModal, setIsConfirmationModal] = useState(true);
   const [typeStyle, setTypeStyle] = useState('default');
 
-  const getTrainers = async () => {
-    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers`);
-    const data = await response.json();
-    setTrainers(data.data);
-  };
+  const dispatch = useDispatch();
+  const trainers = useSelector((state) => state.trainers.data);
+  const [isLoading, setIsLoading] = useState(true);
+
   useEffect(() => {
-    getTrainers();
-  });
+    setTimeout(() => {
```

Comment:
What is the function of that timeout there?

Reply:
This time out is set to force the time out so that the spinner can be displayed easily.
I left it to work better but this must be removed before merging.